sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -143,8 +143,18 @@ private[sql] object FileSourceStrategy extends Strategy with Logging {
           val defaultParallelism = files.sparkSession.sparkContext.defaultParallelism
           val totalBytes = selectedPartitions.flatMap(_.files.map(_.getLen + openCostInBytes)).sum
           val bytesPerCore = totalBytes / defaultParallelism
-          val maxSplitBytes = Math.min(defaultMaxSplitBytes,
-            Math.max(openCostInBytes, bytesPerCore))
+          val maxSplitBytes = {
+            // Since `LineRecordReader` in hadoop cannot split files compressed by some codecs
+            // (e.g., gzip), check if all the input files are splittable here.
+            if (files.fileFormat.canSplitFiles(
```

Comment:
I think we should check the file whether it's splittable or not one by one, do not split it if it's not.